### PR TITLE
Bug 1938903: Monitoring dashboards: Add "Custom time range" modal

### PR DESCRIPTION
**Pull request by kyoto**
Allow setting any custom time range for the dashboards. Uses a modal for
inputting the start and end times for the range.

screenshot

FYI @cshinn

### CODE REVIEW

**Review by spadgett — APPROVED** [review]
/lgtm
/hold for QE/PX/docs approval

**Comment by kyoto** [comment]
This also fixes bug https://bugzilla.redhat.com/show_bug.cgi?id=1938903

**Comment by lihongyan1** [comment]
Test with branch, filed the following bug with target release 4.8-premerge
Bug 1952741 - Monitoring DB: after set Time Range as Custom time range, no data display

**Comment by kyoto** [comment]
Force pushed a change to address bug [1952741](https://bugzilla.redhat.com/show_bug.cgi?id=1952741) (CC @lihongyan1)

Also rebased

**Review by spadgett — APPROVED** [review]
/lgtm

**Comment by lbarbeevargas** [comment]
@kyoto What does the user select to access the Custom time range modal? So the step before the modal opens? I'm adding this option to the docs:
[Reviewing monitoring dashboards as a cluster administrator](https://docs.openshift.com/container-platform/4.7/monitoring/reviewing-monitoring-dashboards.html#reviewing-monitoring-dashboards-admin_reviewing-monitoring-dashboards)
[Reviewing monitoring dashboards as a developer](https://docs.openshift.com/container-platform/4.7/monitoring/reviewing-monitoring-dashboards.html#reviewing-monitoring-dashboards-developer_reviewing-monitoring-dashboards)
Thanks!

**Comment by kyoto** [comment]
@lbarbeevargas To access the modal, you first open the "Time Range" dropdown, then select "Custom time range".

**Comment by kyoto** [comment]
Rebased and fixed a date formatting issue caused by conflicting with https://github.com/openshift/console/pull/8734

File: frontend/public/components/monitoring/dashboards/custom-time-range-modal.tsx [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+            <label>{t('public~From')}</label>
+          </div>
+          <div className="col-sm-4">
+            <DatePicker
+              onChange={(str, date) => setFromDate(date)}
+              value={dateFormatter.format(fromDate)}
+            />
+          </div>
+          <div className="col-sm-4">
+            <TimePicker defaultTime={fromTime} is24Hour onChange={setFromTime} />
```
Comment (spadgett) [inline frontend/public/components/monitoring/dashboards/custom-time-range-modal.tsx:61]: Ideally, `is24Hour` would only be set for locales that use 24 hour time, but I don't see a good API to check. (And it looks like that might add some additional logic, too, if we're getting back a string.)

**Review by spadgett — APPROVED** [review]
/lgtm

**Review by spadgett — APPROVED** [review]
/lgtm

**Pull request merged**